src/Core/Grid/DataProvider/LogGridDataProvider.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\DataProvider;
+
+use Doctrine\DBAL\Driver\Connection;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShopBundle\Entity\Repository\LogRepository;
+use PrestaShopBundle\Service\Hook\HookDispatcher;
+
+/**
+ * Class LogsGridDataProvider is responsible for providing data for Logs grid
+ */
+final class LogGridDataProvider implements GridDataProviderInterface
```

Comment:
@mickaelandrieu  I think i'd like to change data provider's interface to just one method before merging it. 
I imagine it to be something like this:

```php
    public function getData(SearchCriteriaInterface $searchCriteria)
    {
        // some dedicated service to build search & count queries (or maybe we can build only one query?)
        // OPTION 1: and maybe dispatch hook inside query builder to allow modify them
        $searchQuery = $this->queryBuilder->buildSearchQuery($searchCriteria);
        $countQuery = $this->queryBuilder->buildCountQuery($searchCriteria);

        // OPTION 2: or dispatch hook here. In this case, maybe we should pass grid definition as well?
        $this->hookDispatcher->dispatchWithParams('modifyQuery', [
            'search_query' => $searchQuery,
            'count_query' => $countQuery,
        ]);

        $records = $searchQuery->execute()->fetchAll(\PDO::FETCH_ASSOC);    // filtered records
        $recordsTotal = $countQuery->execute()->fetch(); // just get total count of filtered records

        // final grid data ready for rendering. Raw query is needed for SQL Manager
        return new GridData($records, $recordsTotal, $searchQuery->getSQL());
    }
```
In this case, each new grid would have to implement 2 contracts: query builder & definition, which i think makes sense? Or not? Let me know you think.

Reply:
What is the difference between GridDataProviders and GridData?

What problem(s) the addition of GridData objects and LogGridDataProviderInterface::getData is trying to solve?

Reply:
GridData would be just a value object containing final grid data that is ready for rendering. 

At the moment GridDataProvider returns not only data, but also query (getQuery() method) and it seems that we will need 2 queries later on (one for filtered rows and another for filtered rows count?). In this case we could have single data provider that depends on some $queryBuilder interface, in the end we would have one data provider and multiple $queryBuilders. Do you think it's worth doing so or maybe there's a better way?

Reply:
as far as I can see, GridData and GridDataProvider have the same methods, so they have the same responsibilities: am I right?

Reply:
yes, sames methods, but GridData is responsible just for holding grid data while DataProvider is now responsible for retrieving rows, rows count & query. So we could make data provider responsible just for returning final grid data.
However, if you think current implementation is good for now, lets keep it so we can improve it later while migrating other controllers. :)

Reply:
> So we could make data provider responsible just for returning final grid data.

I like this idea, but this means the only public method in ``GridDataProviderIntefrace`` should be ``getGridData() : GridDataInterface``.

It's late, I'll review a little bit more tomorrow :)

Thanks for your answers, good night !

Reply:
yes, exactly. we leave only one method ```getData($searchParams);``` on data provider interface.

let me know your ideas on this. :)